edge-util/src/Microsoft.Azure.Devices.Edge.Util/VerifyTwinSignature.cs

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Util
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Security.Cryptography;
+    using System.Security.Cryptography.X509Certificates;
+
+    using Org.Webpki.JsonCanonicalizer;
+    public class VerifyTwinSignature
+    {
+        public static bool VerifyModuleTwinSignature(string desiredProperties, string header, byte[] signatureBytes, X509Certificate2 signerCert, string algorithmScheme, HashAlgorithmName hashAlgorithm)
+        {
+            try
+            {
+                JsonCanonicalizer canonicalizerDesiredProperties = new JsonCanonicalizer(desiredProperties);
+                byte[] canonicalizedDesiredProperties = canonicalizerDesiredProperties.GetEncodedUTF8();
+
+                JsonCanonicalizer canonicalizerHeader = new JsonCanonicalizer(header);
+                byte[] canonicalizedHeader = canonicalizerHeader.GetEncodedUTF8();
+
+                byte[] canonicalizedCombinedDesiredProperties = new byte[canonicalizedHeader.Length + canonicalizedDesiredProperties.Length];
+
+                Array.Copy(canonicalizedHeader, 0, canonicalizedCombinedDesiredProperties, 0, canonicalizedHeader.Length);
+                Array.Copy(canonicalizedDesiredProperties, 0, canonicalizedCombinedDesiredProperties, canonicalizedHeader.Length, canonicalizedDesiredProperties.Length);
+
+                if (algorithmScheme == "ES")
+                {
+                    ECDsa eCDsa = signerCert.GetECDsaPublicKey();
+                    return eCDsa.VerifyData(canonicalizedCombinedDesiredProperties, signatureBytes, hashAlgorithm);
+                }
+                else if (algorithmScheme == "RS")
+                {
+                    RSA rsa = signerCert.GetRSAPublicKey();
+                    RSASignaturePadding rsaSignaturePadding = RSASignaturePadding.Pkcs1;
+                    return rsa.VerifyData(canonicalizedCombinedDesiredProperties, signatureBytes, hashAlgorithm, rsaSignaturePadding);
+                }
+                else
+                {
+                    throw new TwinSignatureAlgorithmException("DSA Algorithm Type not supported");
+                }
+            }
+            catch (Exception ex)
+            {
+                throw ex;
+            }
+        }
+
+        public static KeyValuePair<string, HashAlgorithmName> CheckIfAlgorithmIsSupported(string dsaAlgorithm)
```

Comment:
Nit: Rename to GetAlgorithm or ParseAlgorithm.

Reply:
Also, do we need the algorithmScheme? Can't we infer it from the Algorithm itself? 

If you still want it for some reason, maybe make it an Enum as well..

Reply:
The supported algorithms are made as enum and it contains two parts - the algorithm Scheme such as EC, RS and the Hash algorithm. Once we parse the enum, we get the key value pair and use that for the verification process. I will rename to `ParseAlgorithm`